File: app/src/main/java/com/jmalinen/blackjack/ui/components/ExtraPlayersArea.kt
package com.jmalinen.blackjack.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.jmalinen.blackjack.model.ExtraPlayerState

@Composable
fun ExtraPlayersArea(
    extraPlayers: List<ExtraPlayerState>,
    modifier: Modifier = Modifier
) {
    if (extraPlayers.isEmpty()) return

    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 2.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceEvenly
        ) {
            extraPlayers.forEachIndexed { index, ep ->
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text(
                        text = "P${index + 1}",
                        color = Color.White.copy(alpha = 0.5f),
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Bold
                    )
                    HandView(
                        hand = ep.hand,
                        isActive = false,
                        showScore = ep.hand.cards.isNotEmpty(),
                        result = ep.result,
                        cardScale = 0.75f
                    )
                }
            }
        }
    }
}

File: ios/Blackjack/Blackjack/Views/Components/ExtraPlayersArea.swift
import SwiftUI

struct ExtraPlayersArea: View {
    let extraPlayers: [ExtraPlayerState]
    var cardScale: CGFloat = 0.75

    var body: some View {
        if extraPlayers.isEmpty { return AnyView(EmptyView()) }

        return AnyView(
            VStack {
                HStack(spacing: 0) {
                    ForEach(Array(extraPlayers.enumerated()), id: \.offset) { index, ep in
                        VStack(spacing: 0) {
                            Text("P\(index + 1)")
                                .font(.system(size: 11, weight: .bold))
                                .foregroundColor(.white.opacity(0.5))
                            HandView(
                                hand: ep.hand,
                                isActive: false,
                                showScore: !ep.hand.cards.isEmpty,
                                result: ep.result,
                                cardScale: cardScale
                            )
                        }
                    }
                }
                .frame(maxWidth: .infinity)
            }
            .padding(.horizontal, 16)
        )
    }
}
